### PR TITLE
Fixed obvious unnecessary users of duration.

### PR DESCRIPTION
**Pull request by Ducasse**
- Next step the valueWithin: onTimeout:

### CODE REVIEW

**Comment by Ducasse** [comment]
Hi @guillep
I will wait for your review before merging because it is a bit sensitive

File: src/Announcements-Core-Tests/AnnouncerTest.class.st
Comment (guillep) [inline src/Announcements-Core-Tests/AnnouncerTest.class.st:96]: This is not equivalent. 1 second is 1000 milliseconds :)

File: src/DrTests-TestsProfiling/DTTestsProfilerVisitor.class.st
Comment (guillep) [inline src/DrTests-TestsProfiling/DTTestsProfilerVisitor.class.st:43]: here I don't see why it would be bad to use `timeToRun`. I don't think  `timeToRun` is bad per-se.

Here for example, we already need a duration, so we are not introducing an extra dependency and the code reads nicer...

File: src/Kernel-CodeModel/BlockClosure.class.st
Comment (guillep) [inline src/Kernel-CodeModel/BlockClosure.class.st:391]: Why here we don't use `Time`?

File: src/Kernel-CodeModel/BlockClosure.class.st
Comment (guillep) [inline src/Kernel-CodeModel/BlockClosure.class.st:603]: Same, why not having a dependency to Time here?

File: src/Kernel-CodeModel/BlockClosure.class.st
Comment (guillep) [inline src/Kernel-CodeModel/BlockClosure.class.st:395]: And maybe this method should belong to the same package as `BlockClosure>>bench` (I'd create maybe later a new package?)

File: src/Kernel-CodeModel/BlockClosure.class.st
Comment (guillep) [inline src/Kernel-CodeModel/BlockClosure.class.st:1032]: I'm not convinced this method should be in Kernel :) But I understand it was before like this already.

File: src/Kernel-Tests-Extended/ValueWithinFixTest.class.st
Comment (guillep) [inline src/Kernel-Tests-Extended/ValueWithinFixTest.class.st:12]: haha :)

File: src/Kernel-Tests-Extended/ValueWithinFixTest.class.st
Comment (guillep) [inline src/Kernel-Tests-Extended/ValueWithinFixTest.class.st:39]: Super strange to ask the clock to the closure.

File: src/Kernel/Delay.class.st
Comment (guillep) [inline src/Kernel/Delay.class.st:67]: Hmm, shouldn't this use the constants in `ChronologyConstants`? It makes sense that `Delay` depends on that

File: src/Kernel/Delay.class.st
Comment (guillep) [inline src/Kernel/Delay.class.st:80]: same here

File: src/Kernel/Delay.class.st
Comment (guillep) [inline src/Kernel/Delay.class.st:96]: and here

File: src/SUnit-Core/TestAsserter.class.st
Comment (guillep) [inline src/SUnit-Core/TestAsserter.class.st:534]: ;)

**Review by guillep — APPROVED** [review]
I have some comments to discuss!

File: src/Announcements-Core-Tests/AnnouncerTest.class.st
Comment (Ducasse) [reply src/Announcements-Core-Tests/AnnouncerTest.class.st:96]: :) I will fix this.

File: src/DrTests-TestsProfiling/DTTestsProfilerVisitor.class.st
Comment (Ducasse) [reply src/DrTests-TestsProfiling/DTTestsProfilerVisitor.class.st:43]: I wanted to deprecate timeToRun and to make explicit the dependency from DrTest to the future Duration package.

File: src/Kernel-CodeModel/BlockClosure.class.st
Comment (Ducasse) [reply src/Kernel-CodeModel/BlockClosure.class.st:391]: I thought that we want to get Time out of the kernel.
So I copied the primitive there.

File: src/Kernel-CodeModel/BlockClosure.class.st
Comment (Ducasse) [reply src/Kernel-CodeModel/BlockClosure.class.st:603]: I thought that we do not want to have a dependency to Time from the kernel.

File: src/Kernel-Tests-Extended/ValueWithinFixTest.class.st
Comment (Ducasse) [reply src/Kernel-Tests-Extended/ValueWithinFixTest.class.st:39]: Indeed. Now this is a test of closure and if we want to run these tests without the time package then I do not see how we can do it.

File: src/Kernel/Delay.class.st
Comment (Ducasse) [reply src/Kernel/Delay.class.st:67]: Yes it would be better.  Now we can do it in a second turn. Because it would be good to have it run and then create extra package.

**Comment by Ducasse** [comment]
I feel like I will need help. I tried to repair and pharo disappeared - not sure that I can reproduce it. I think that the fact that I share repo is making things more complex when working on long branches. I have to repair each time pharo changes or that I work on Pharo. I will change this set up because it brings a lot of noise and pain. Tell me if I'm wrong.

File: src/DrTests-TestsProfiling/DTTestsProfilerVisitor.class.st
Comment (guillep) [reply src/DrTests-TestsProfiling/DTTestsProfilerVisitor.class.st:43]: But I like timeToRun!!! Just not in the kernel!

File: src/Kernel-CodeModel/BlockClosure.class.st
Comment (guillep) [reply src/Kernel-CodeModel/BlockClosure.class.st:395]: @Ducasse to me the answer to your comments above is here. I want timeToRun & co out of the kernel too. They should be packaged elsewhere, maybe together with Time in first instance.

File: src/Kernel-Tests-Extended/ValueWithinFixTest.class.st
Comment (guillep) [reply src/Kernel-Tests-Extended/ValueWithinFixTest.class.st:39]: Maybe value with timeout does not need to be kernel? How is it used? I'll check

File: src/DrTests-TestsProfiling/DTTestsProfilerVisitor.class.st
Comment (Ducasse) [reply src/DrTests-TestsProfiling/DTTestsProfilerVisitor.class.st:43]: Here it was really syntactical sugar and make the dependency analysis more difficult. 
This is why I switch it. And it does not remove the dependency of DrTest to Duration.

File: src/Kernel-CodeModel/BlockClosure.class.st
Comment (Ducasse) [reply src/Kernel-CodeModel/BlockClosure.class.st:395]: my goal was to have timeToRun in a Duration Package that is on top of Time's package.

File: src/Kernel-Tests-Extended/ValueWithinFixTest.class.st
Comment (Ducasse) [reply src/Kernel-Tests-Extended/ValueWithinFixTest.class.st:39]: Yes this too. Now having a version with milliseconds was easy and I like to have it this way people can choose what they want and duration is optional.

**Comment by Ducasse** [comment]
On one of the three platforms the tests fully pass.

File: src/DrTests-TestsProfiling/DTTestsProfilerVisitor.class.st
Comment (tinchodias) [reply src/DrTests-TestsProfiling/DTTestsProfilerVisitor.class.st:43]: I agree with guille in timeToRun

**Comment by tinchodias** [comment]
In Bloc "ecosystem" I see 21 methods sending it.

I used `Time millisecondClockValue` in benchmark suites, the pattern in general is:
```
startMS := Time millisecondClockValue.
"do things"
durationMS := Time millisecondClockValue - startMS.
```
but the durationMS often happens in other method.

File: src/DrTests-TestsProfiling/DTTestsProfilerVisitor.class.st
Comment (Ducasse) [reply src/DrTests-TestsProfiling/DTTestsProfilerVisitor.class.st:43]: Me too. Now I repeat it because may be I was unclear. 
In DrTest we can have a dependency to Duration but 
- I wanted to make this explicit hence I refer to the class Duration
- I wanted to make sure that we do not use timeToRun in BlockClosure.

**Comment by Ducasse** [comment]
My point is that often duration is often not useful because it just put a millisecond number into a newly created duration and from that duration object we retrieve the milliseconds!

**Comment by Ducasse** [comment]
@guillep I do not know how to resolve the conflict. 
Now I'm thinking that once the conflict is resolved we can merge and we can address your question in another PR.

**Comment by guillep** [comment]
It must be resolved in iceberg or in the command line... The thing is that you touched Duration while another PR (mine?) moved it to another package. So the github UI will not help there...

**Comment by Ducasse** [comment]
I merge now and we can do another iteration to handle guille's comments.

**Pull request merged**
